Add install for Microsoft Azure

tomkerkhove opened the pull request:
### Summary

Add installation guidance for Microsoft Azure.

Currently only how you can setup a PostgreSQL DB and run Kong on Azure Container Instances, but more will follow later on.

### Full changelog

* Added Microsoft Azure to known distributions
* Added guidance for installation on Azure

### Issues resolved

Relates to https://github.com/Kong/kong/issues/3941

<!-- Have you done all of these things?  -->
### Checklist:
<!-- add "N/A" to the end of each line that's irrelevant to your changes -->
<!-- to check an item, place an "x" in the box like so: "- [x] Documentation" -->
- [X] [Commit message & atomicity](https://github.com/Kong/docs.konghq.com/blob/master/CONTRIBUTING.md#commit-atomicity) checked
- [ ] Documentation <!-- Adding a new feature? Do you need to document it the README.md or otherwise? --> N/A
- [X] Spellchecked my updates
- [X] Ready to be merged <!-- In your opinion, is this ready to be merged as soon as it's reviewed? -->

<!-- feel free to add additional comments -->

coopr commented:
Great start @tomkerkhove thanks! 

It'd be awesome if you could at include a mention of https://azuremarketplace.microsoft.com/en-us/marketplace/apps/bitnami.kong and https://azuremarketplace.microsoft.com/en-us/marketplace/apps/bitnami.kong-cluster - I know you probably aren't that familiar with them, so no need to say much - me and/or the Bitnami folks will add some details after we get your PR merged.

coopr commented on app/install/microsoft-azure.md:
```diff
@@ -0,0 +1,99 @@
+  <div class="text-center">
+    <strong>Note</strong>: See <a href="https://github.com/Kong/docker-kong/issues/188" target="blank">#188</a> for more information.
+  </div>
+</div>
```
Needs a blank line at the end.

tomkerkhove replied:
I added a blank line but seems like the diff does not show it at the moment.

coopr commented on app/install/microsoft-azure.md:
```diff
@@ -0,0 +1,99 @@
+1. [Running PostgreSQL on Azure with Azure Database for PostgreSQL](#running-postgresql-on-azure-with-azure-database-for-postgresql)
+1. [Running Cassandra on Azure with Azure Cosmos Db](#running-cassandra-on-azure-with-azure-cosmos-db)
+
+# Running Kong on Azure Container Instances
```
Should be `##` - `title: Install - Microsoft Azure` in the front matter is rendered as the page title and the sole H1 header on the page - all other headers must start at H2, to avoid having more than H1 on the page.

coopr commented on app/install/microsoft-azure.md:
````diff
@@ -0,0 +1,99 @@
+
+1. **Run the migrations**
+
+    ```bash
````
@darrenjennings I was thinking I'd see the awesomesauce from https://github.com/Kong/docs.konghq.com/pull/969 here, but when I run the docs site locally, and visit this page, and triple-click on the code sample, the leading $ prompt gets selected - any ideas?

![select-prompt](https://user-images.githubusercontent.com/94904/47884673-ea708e00-ddee-11e8-971e-400db2161f63.gif)

tomkerkhove commented:
My pleasure @coopr! I've briefly added them and kept the section a bit more general for when new solution templates will be added but would be good if the Bitnami folks could jump in and provide a better description of what to expect and maybe even expand the information in the marketplace a bit more given it is not 100% clear what to expect, certainly for the "Kong Certified" one in my opinion.

Later on, I will add more examples of how to run them on things like Azure Service Fabric Mesh, but this will only be later on when I do this as part of https://github.com/tomkerkhove/containo but should not be that hard.

tomkerkhove commented:
Feel free to include me in all Microsoft Azure issues/PRs to review or provide input

tomkerkhove commented:
I've noticed that GCP has one page for all their docs and AWS has seperate pages per service.

What is the preferred approach? Given other services will follow such as Azure Service Fabric Mesh, is it best to use multiple install pages then?

Tom Kerkhove
Azure Architect at Codit, Microsoft Azure MVP

From: Cooper Marcus
Sent: Thursday, November 1, 23:59
Subject: Re: [Kong/docs.konghq.com] Add install for Microsoft Azure (#1018)
To: Kong/docs.konghq.com
Cc: Tom Kerkhove, Mention



@coopr requested changes on this pull request.

Great start!

________________________________

In app/install/microsoft-azure.md<https://github.com/Kong/docs.konghq.com/pull/1018#discussion_r230223692>:

> +1. Create a new database called "kong" by using your favorite tool
+
+<div class="alert alert-warning">
+  <div class="text-center">
+    <strong>Note</strong>: Before connecting to your new database, make sure your IP address is whitelisted in "Connection Security"
+  </div>
+</div>
+
+# Running Cassandra on Azure with Azure Cosmos Db
+Currently, Azure Cosmos Db is not supported as a Cassandra data store.
+
+<div class="alert alert-info">
+  <div class="text-center">
+    <strong>Note</strong>: See <a href="https://github.com/Kong/docker-kong/issues/188" target="blank">#188</a> for more information.
+  </div>
+</div>


Needs a blank line at the end.

________________________________

In app/install/microsoft-azure.md<https://github.com/Kong/docs.konghq.com/pull/1018#discussion_r230223960>:

> +id: page-install-method
+title: Install - Microsoft Azure
+header_title: Microsoft Azure
+header_icon: /assets/images/icons/icn-installation.svg
+breadcrumbs:
+  Installation: /install
+---
+
+## Table of Contents
+
+<!-- FIXME the list below should be an unordered list, but currently those do not render correctly in this section of the Docs site - depends on https://github.com/Kong/docs.konghq.com/issues/917 -->
+1. [Running Kong on Azure Container Instances](#running-kong-on-azure-container-instances)
+1. [Running PostgreSQL on Azure with Azure Database for PostgreSQL](#running-postgresql-on-azure-with-azure-database-for-postgresql)
+1. [Running Cassandra on Azure with Azure Cosmos Db](#running-cassandra-on-azure-with-azure-cosmos-db)
+
+# Running Kong on Azure Container Instances


Should be ## - title: Install - Microsoft Azure in the front matter is rendered as the page title and the sole H1 header on the page - all other headers must start at H2, to avoid having more than H1 on the page.

________________________________

In app/install/microsoft-azure.md<https://github.com/Kong/docs.konghq.com/pull/1018#discussion_r230226697>:

> +# Running Kong on Azure Container Instances
+<a href="https://docs.microsoft.com/en-us/azure/container-instances/container-instances-overview" target="blank">Azure Container Instances</a> is a great way to run lightweight containers in a serverless fashion.
+
+Running Kong on Azure Container Instances is super easy:
+
+1. **Provision a data store**
+
+    Provision the data store that you want to use:
+    1. [Running Cassandra on Azure with Azure Cosmos Db](#running-cassandra-on-azure-with-azure-cosmos-db)
+    1. [Running PostgreSQL on Azure with Azure Database for PostgreSQL](#running-postgresql-on-azure-with-azure-database-for-postgresql)
+
+1. **Open the Cloud Shell or Azure CLI**
+
+1. **Run the migrations**
+
+    ```bash


@darrenjennings<https://github.com/darrenjennings> I was thinking I'd see the awesomesauce from #969<https://github.com/Kong/docs.konghq.com/pull/969> here, but when I run the docs site locally, and visit this page, and triple-click on the code sample, the leading $ prompt gets selected - any ideas?

[select-prompt]<https://user-images.githubusercontent.com/94904/47884673-ea708e00-ddee-11e8-971e-400db2161f63.gif>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Kong/docs.konghq.com/pull/1018#pullrequestreview-170914237>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AEJPP5O7MJ3aOvWkz2D4DvJ0cJHJ11GJks5uq3zHgaJpZM4YJ2g7>.

coopr commented:
> I've noticed that GCP has one page for all their docs and AWS has seperate pages per service. What is the preferred approach? Given other services will follow such as Azure Service Fabric Mesh, is it best to use multiple install pages then?

Great catch, right you are!

I see that Amazon essentially has three install pages:

1. https://docs.konghq.com/install/aws-linux/
1. https://docs.konghq.com/install/aws-marketplace/
1. https://docs.konghq.com/install/aws-cloudformation/

While, for example, Kubernetes and Google each have one page, though they could really have multiple pages each:

- https://docs.konghq.com/install/kubernetes/
- https://docs.konghq.com/install/google-cloud/

Frankly, our whole approach to structuring these pages needs reconsideration (FYI @steph-kalacis ) - not only are those ^^ pages inconsistent, but we have new and detailed documentation that doesn't really have a place within the current `/install/` structure like https://github.com/Kong/kubernetes-ingress-controller#documentation

My advice is this: For now, let's stuff all the Azure-related docs into a single page, like we've done with (most of) the Kubernetes docs. Then, later, when we have time to consider, decide, and implement a better approach, we'll refactor all the too-full pages into a structure that makes sense. Sound good?

tomkerkhove commented:
That sounds good to me, we'll only need to make sure that we don't break URLs as far as we can.

Tom Kerkhove
Azure Architect at Codit, Microsoft Azure MVP

________________________________
From: Cooper Marcus <notifications@github.com>
Sent: Friday, November 2, 2018 11:25:44 PM
To: Kong/docs.konghq.com
Cc: Tom Kerkhove; Mention
Subject: Re: [Kong/docs.konghq.com] Add install for Microsoft Azure (#1018)


@tomkerkhove<https://github.com/tomkerkhove> I've been thinking about this addition, and I've realized we need to move it to a different section of the website.

While it is technically possible to add new sub-pages to /install, I'm reminded that everything on that page is something that Kong Inc. officially supports - and the great contribution you are making isn't one we can commit to supporting (at least not yet).

So, I need to ask that we refactor this content into a Kong Hub addition - I imagine we'd add one or more listings to the Deployment section of https://docs.konghq.com/hub/

To do so, I'll point you towards https://github.com/Kong/docs.konghq.com/blob/master/CONTRIBUTING.md#contributing-to-kong-documentation-and-the-kong-hub - and to say sorry for leading you astray at first, please let me know if you'd like me to make the first quick translation from /install to /hub 😊

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Kong/docs.konghq.com/pull/1018#issuecomment-435527008>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AEJPP7vxYvns5OYnA44EuPB_LaA6HdaXks5urMZogaJpZM4YJ2g7>.

coopr commented:
Sorry I think maybe I wasn't clear - we aren't going to merge this as is, and instead we need to change the addition of Azure installation methods from being listed under `/install` to being listed under `/hub` - and the method for adding things under `/hub` is different than adding things under `/install` - thus, this PR needs changes, or we need to close it and start a new one, to list the new Azure options under `/hub` - does that make sense?

coopr commented:
@tomkerkhove I've added a commit to this PR that starts the changes I'm suggesting - the new `app/_hub/tomkerkhove/microsoft_azure/index.md` page needs tuning-up - can you give that a shot?

tomkerkhove commented:
Thanks for the update @coopr, I've clearly misread what you said - Sorry for that.
I'll have a look at the new page!

tomkerkhove commented:
Updated the PR with a proposal for deployment docs in Hub. I've created a general one for Microsoft Azure that includes marketplace and one specific to Azure Container Instances.

Hub overview:
![screenshot_1](https://user-images.githubusercontent.com/4345663/47998229-daaebd80-e0fe-11e8-886b-25bb3209dc91.png)

General Azure:
![screenshot_2](https://user-images.githubusercontent.com/4345663/47998230-daaebd80-e0fe-11e8-984a-e035eaf93df9.png)

Azure Container Instances:
![screenshot_3](https://user-images.githubusercontent.com/4345663/47998233-db475400-e0fe-11e8-895e-a3ab03306e89.png)

Linking this to https://github.com/tomkerkhove/kong-deployment-on-azure/issues/3 & https://github.com/tomkerkhove/kong-deployment-on-azure/issues/1 and more coming later on such as https://github.com/tomkerkhove/kong-deployment-on-azure/issues/2.

Are we good to go @coopr ?

coopr commented on app/_hub/tomkerkhove/microsoft_azure_container_instances/index.md:
```diff
@@ -0,0 +1,222 @@
+  This guide walks you through deploying and running Kong on Microsoft Azure Container Instances.
+
+  We will be using either Azure Database for PostgreSQL or Cosmos Db to store the gateway configuration.
+support_url: # (Optional) A specific URL of your own for this extension.
```
Whoops, this isn't actually optional - I need to update the template - it should be https://github.com/tomkerkhove/kong-deployment-on-azure/issues I think

tomkerkhove replied:
Good point :+1:

coopr commented on app/_hub/tomkerkhove/microsoft_azure_container_instances/index.md:
```diff
@@ -0,0 +1,222 @@
+
+kong_version_compatibility: # required
+  community_edition: # optional
+    compatible:
```
I think this compatibility statement isn't correct - I'm guessing your solution is compatible (eg. tested and known to work with) Kong 0.14.x - if that is correct, please put all the older versions in the `incompatible: ` section below, thanks.

coopr commented on app/_hub/tomkerkhove/microsoft_azure_container_instances/index.md:
```diff
@@ -0,0 +1,222 @@
+    #incompatible:
+  enterprise_edition: # optional
+    compatible:
+        - 0.34-x
```
Same comment as above - if you've not confirmed compatibility, either mark these as incompatible, or comment it out for "unknown compatibility"

coopr commented on app/_hub/tomkerkhove/microsoft_azure_container_instances/index.md:
```diff
@@ -0,0 +1,222 @@
+        - 0.32-x
+    #incompatible:
+
+# EXAMPLE kong_version_compatibility blocks - these examples show how to indicate various compatibilities. Also see other extension files in _app/_hub/ for more examples
```
Feel free to delete all these comments from your PR

coopr commented on app/_hub/tomkerkhove/microsoft_azure/index.md:
```diff
@@ -0,0 +1,158 @@
+kong_version_compatibility: # required
+  community_edition: # optional
+    compatible:
+        - 0.14.x
```
Same comments here as the compatibility section in the other file - I think it needs adjusting.

coopr commented on app/_hub/tomkerkhove/microsoft_azure/index.md:
```diff
@@ -0,0 +1,158 @@
+  #   industry. Lorem Ipsum has been the industry's standard dummy text ever
+  #   since the 1500s.
+
+support_url: # (Optional) A specific URL of your own for this extension.
```
This needs a URL - I'm not clear what it would be - maybe some Bitnami support URL or ??

tomkerkhove replied:
Good point :+1: - Will add one to my repo, can still point them to Bitnami if it's for the solution templates (can grow in future)

tomkerkhove commented:
Everything is cleaned up now, added link for support and removed supported version samples.

However, I'd prefer to keep the other comments around in case I want to change something in the future that I have some guidance what is expected.

coopr commented:
Somehow this PR was breaking another extension - I've added a fix for that.

I need to do one quick review with a colleague, but I think this is good to go - I hope to merge and deploy this soon!

tomkerkhove commented:
Awesome, thanks! No worries, take your time!

tomkerkhove commented:
My pleasure!
